### PR TITLE
Update for plugin to be compilable.

### PR DESCRIPTION
**Pull request by makiam**
Delete uncompilable and unused CenteredBscrollPane class
Update TapPreviewWidget to implement missed getAllViews() method
Disabled with stubs calls to missed Material and Texture classes
Update plugin descriptor to new extensions.xml

### CODE REVIEW

File: src/artofillusion/tapDesigner/TaPDMetalTheme.java [hunk 1]
```diff
@@ -1,79 +0,0 @@
-/*
```
Comment (Sailsman63) [inline src/artofillusion/tapDesigner/TaPDMetalTheme.java:1]: Probably shouldn't delete this file quite yet. It is referenced in `tapDesigner/TaPDPreferences`. Does it actually cause compilation issues?

File: src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java [hunk 2]
```diff
@@ -146,12 +148,13 @@ private void doCancel()
   
   private void doNewMaterial()
   {
-    MaterialsDialog.showNewMaterialWindow(this, sc);
+      JOptionPane.showMessageDialog(this.component, "MaterialsDialog class and showNewMaterialWindow method deleted.");
```
Comment (Sailsman63) [inline src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java:151]: As a short-term fix, these should probably just create a `new TexturesAndMaterialsDialog(this, sc);`

We can clean up the extra UI elements later.

File: src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java [hunk 3]
```diff
@@ -146,12 +148,13 @@ private void doCancel()
...
   }
   
   private void doEditMaterials()
   {
-    sc.showMaterialsDialog(fr);
+    JOptionPane.showMessageDialog(this.component, "Scene's showMessageDialog method deleted.");
```
Comment (Sailsman63) [inline src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java:156]: And same.

File: src/artofillusion/tapDesigner/TaPDObjectTextureDialog.java [hunk 4]
```diff
@@ -358,12 +361,13 @@ private void doEditMapping()
   
   private void doNewTexture()
   {
-    TexturesDialog.showNewTextureWindow(this, sc);
+    JOptionPane.showMessageDialog(this.component, "TexturesDialog class and showNewTextureWindow method deleted.");
```
Comment (Sailsman63) [inline src/artofillusion/tapDesigner/TaPDObjectTextureDialog.java:364]: Same

File: src/artofillusion/tapDesigner/TaPDObjectTextureDialog.java [hunk 5]
```diff
@@ -358,12 +361,13 @@ private void doEditMapping()
...
   }
   
   private void doEditTextures()
   {
-    sc.showTexturesDialog(fr);
+    JOptionPane.showMessageDialog(this.component, "Scene's showTexturesDialog method parameters not match to Frame.");
```
Comment (Sailsman63) [inline src/artofillusion/tapDesigner/TaPDObjectTextureDialog.java:369]: And, last one...

File: src/artofillusion/tapDesigner/TapProcedure.java [hunk 6]
```diff
@@ -1082,8 +1078,8 @@ public void importTextures( BFrame parentFrame )
      */
     public void manageTextures( BFrame parentFrame )
     {
-        theScene.showTexturesDialog( parentFrame );
-        setModified( true );
+        //theScene.showTexturesDialog( parentFrame );
```
Comment (Sailsman63) [inline src/artofillusion/tapDesigner/TapProcedure.java:1081]: These can probably be similar to the ones in the Texture and Material manager dialogs.

File: src/artofillusion/tapDesigner/TapProcedure.java [hunk 7]
```diff
@@ -1106,8 +1102,8 @@ public void importMaterials( BFrame parentFrame )
      */
     public void manageMaterials( BFrame parentFrame )
     {
-        theScene.showMaterialsDialog( parentFrame );
-        setModified( true );
+        //theScene.showMaterialsDialog( parentFrame );
```
Comment (Sailsman63) [inline src/artofillusion/tapDesigner/TapProcedure.java:1105]: Same.

**Review by Sailsman63 — CHANGES_REQUESTED** [review]

**Comment by Sailsman63** [comment]
Sorry for the *long* wait to look at this. I kind of got distracted by stuff in the main application.

File: src/artofillusion/tapDesigner/TaPDMetalTheme.java (hunk 1 above)
Comment (makiam) [reply src/artofillusion/tapDesigner/TaPDMetalTheme.java:1]: Probably should delete TaPDPreferences class as well as this one. The TaPDPreferences  is not used in code...

File: src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java (hunk 2 above)
Comment (makiam) [reply src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java:151]: AOI's TexturesAndMaterialsDialog required EditingWindow and Scene as constructor parameters so it cannot be directly used here.
... Same for all other comments

File: src/artofillusion/tapDesigner/TaPDMetalTheme.java (hunk 1 above)
Comment (Sailsman63) [reply src/artofillusion/tapDesigner/TaPDMetalTheme.java:1]: Probably. It looks like it only ever played with the theme a bit...

File: src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java (hunk 2 above)
Comment (Sailsman63) [reply src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java:151]: Ahh... that's right. Something else to think about.

I'm not sure why TaPD does all of this extra management of Textures and Materials, I guess we can let that be broken for a bit.

Comment (Sailsman63) [reply src/artofillusion/tapDesigner/TaPDObjectMaterialDialog.java:151]: What's not clear is why TaPD doesn't implement `EditingWindow`... it seems to have taken some trouble to break away from AOI core classes in some places, not sure why.

**Comment by Sailsman63** [comment]
There are a couple more changes that need to be made to get this to actually compile. I'll see if I can get them pushed in a little bit.

**Comment by makiam** [comment]
Minimal change to complie this plugin is update aoijar and lib propeties to set proper paths and replace copying TapD.xml and plugins files with copying extensions.xml

**Pull request merged**